diff --git a/cmake_modules/ThirdpartyToolchain.cmake b/cmake_modules/ThirdpartyToolchain.cmake
index 9e3c47d..0d2696a 100644
--- a/cmake_modules/ThirdpartyToolchain.cmake
+++ b/cmake_modules/ThirdpartyToolchain.cmake
@@ -1155,6 +1155,17 @@ macro(build_arrow)
     set(PARQUET_STATIC_LIB
         "${ARROW_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}parquet${CMAKE_STATIC_LIBRARY_SUFFIX}"
     )
+    set(ARROW_BROTLI_PREFIX
+        "${CMAKE_CURRENT_BINARY_DIR}/arrow_ep-prefix/src/arrow_ep-build/brotli_ep/src/brotli_ep-install")
+    set(ARROW_BROTLI_DEC_STATIC_LIB
+        "${ARROW_BROTLI_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}brotlidec-static${CMAKE_STATIC_LIBRARY_SUFFIX}"
+    )
+    set(ARROW_BROTLI_ENC_STATIC_LIB
+        "${ARROW_BROTLI_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}brotlienc-static${CMAKE_STATIC_LIBRARY_SUFFIX}"
+    )
+    set(ARROW_BROTLI_COMMON_STATIC_LIB
+        "${ARROW_BROTLI_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}brotlicommon-static${CMAKE_STATIC_LIBRARY_SUFFIX}"
+    )
 
     set(ARROW_CMAKE_ARGS
         ${EP_COMMON_CMAKE_ARGS}
@@ -1185,6 +1196,7 @@ macro(build_arrow)
         -DARROW_WITH_ZSTD=ON
         -DARROW_WITH_BZ2=OFF
         -DARROW_WITH_BROTLI=ON
+        -DBrotli_SOURCE=BUNDLED
         -DZSTD_ROOT=${ARROW_ZSTD_ROOT}
         -DZLIB_ROOT=${ARROW_ZLIB_ROOT}
         -DSnappy_ROOT=${ARROW_SNAPPY_ROOT}
@@ -1207,6 +1219,9 @@ macro(build_arrow)
                                          "${PARQUET_STATIC_LIB}"
                                          "${ARROW_DATASET_STATIC_LIB}"
                                          "${ARROW_ACERO_STATIC_LIB}"
+                                         "${ARROW_BROTLI_DEC_STATIC_LIB}"
+                                         "${ARROW_BROTLI_ENC_STATIC_LIB}"
+                                         "${ARROW_BROTLI_COMMON_STATIC_LIB}"
                         DEPENDS zstd
                                 snappy
                                 lz4
@@ -1257,6 +1272,18 @@ macro(build_arrow)
     add_dependencies(arrow_dataset arrow_ep)
     add_dependencies(arrow_acero arrow_ep)
 
+    add_library(brotlidec STATIC IMPORTED)
+    set_target_properties(brotlidec PROPERTIES IMPORTED_LOCATION "${ARROW_BROTLI_DEC_STATIC_LIB}")
+    add_dependencies(brotlidec arrow_ep)
+
+    add_library(brotlienc STATIC IMPORTED)
+    set_target_properties(brotlienc PROPERTIES IMPORTED_LOCATION "${ARROW_BROTLI_ENC_STATIC_LIB}")
+    add_dependencies(brotlienc arrow_ep)
+
+    add_library(brotlicommon STATIC IMPORTED)
+    set_target_properties(brotlicommon PROPERTIES IMPORTED_LOCATION "${ARROW_BROTLI_COMMON_STATIC_LIB}")
+    add_dependencies(brotlicommon arrow_ep)
+
     target_link_libraries(arrow_acero INTERFACE arrow)
 
     target_link_libraries(arrow_dataset INTERFACE arrow_acero)
@@ -1266,6 +1293,9 @@ macro(build_arrow)
                                     snappy
                                     lz4
                                     zlib
+                                    brotlidec
+                                    brotlienc
+                                    brotlicommon
                                     re2::re2
                                     arrow_bundled_dependencies)
 
